File: app/src/main/java/com/commit451/gitlab/api/converter/UriTypeConverter.java
package com.commit451.gitlab.api.converter;

import android.net.Uri;

import com.bluelinelabs.logansquare.typeconverters.StringBasedTypeConverter;

/**
 * Simple Uri type converter
 */
public class UriTypeConverter extends StringBasedTypeConverter<Uri> {

    @Override
    public String convertToString(Uri object) {
        return object.toString();
    }

    @Override
    public Uri getFromString(String string) {
        return Uri.parse(string);
    }
}

File: app/src/main/java/com/commit451/gitlab/api/converter/UriTypeConverter.kt
package com.commit451.gitlab.api.converter

import android.net.Uri

import com.bluelinelabs.logansquare.typeconverters.StringBasedTypeConverter

/**
 * Simple Uri type converter
 */
class UriTypeConverter : StringBasedTypeConverter<Uri>() {

    override fun convertToString(`object`: Uri): String {
        return `object`.toString()
    }

    override fun getFromString(string: String?): Uri? {
        if (string != null) {
            return Uri.parse(string)
        } else {
            return null
        }
    }
}
